### ISSUE TITLE
Some advice on Actor-Critic neural

### ISSUE THREAD

**Issue by RozenAstrayChen** [MorvanZhou/Reinforcement-learning-with-tensorflow#93]
Hi MorvanZhou, thanks for your tutorial !
I found some problem in [CartPole-v0 by Actor-Critic](https://github.com/MorvanZhou/Reinforcement-learning-with-tensorflow/blob/master/contents/8_Actor_Critic_Advantage/AC_CartPole.py) , when I train i found reward is not stable and decrease after 600 episode.

![origin](https://user-images.githubusercontent.com/19503080/48131095-d682cb80-e2c9-11e8-9379-8fb8232a800c.jpg)

 So I change the neural to this:
In __actor__
```
        with tf.variable_scope('Actor'):
            l1 = tf.layers.dense(
                inputs=self.s,
                units=32,  # number of hidden units
                activation=tf.nn.relu,
                kernel_initializer=tf.random_normal_initializer(0.,
                                                                .1),  # weights
                bias_initializer=tf.constant_initializer(0.1),  # biases
                name='l1')

            l2 = tf.layers.dense(
                inputs=l1,
                units=32,  # number of hidden units
                activation=tf.nn.relu,
                kernel_initializer=tf.random_normal_initializer(0.,
                                                                .1),  # weights
                bias_initializer=tf.constant_initializer(0.1),  # biases
                name='l2')

            self.acts_prob = tf.layers.dense(
                inputs=l2,
                units=n_actions,  # output units
                activation=tf.nn.softmax,  # get action probabilities
                kernel_initializer=tf.random_normal_initializer(0.,
                                                                .1),  # weights
                bias_initializer=tf.constant_initializer(0.1),  # biases
                name='acts_prob')
```
__critic__
```
        with tf.variable_scope('Critic'):
            l1 = tf.layers.dense(
                inputs=self.s,
                units=32,  # number of hidden units
                activation=tf.nn.relu,  # None
                # have to be linear to make sure the convergence of actor.
                # But linear approximator seems hardly learns the correct Q.
                kernel_initializer=tf.random_normal_initializer(0.,
                                                                .1),  # weights
                bias_initializer=tf.constant_initializer(0.1),  # biases
                name='l1')

            l2 = tf.layers.dense(
                inputs=l1,
                units=32,  # number of hidden units
                activation=tf.nn.relu,  # None
                # have to be linear to make sure the convergence of actor.
                # But linear approximator seems hardly learns the correct Q.
                kernel_initializer=tf.random_normal_initializer(0.,
                                                                .1),  # weights
                bias_initializer=tf.constant_initializer(0.1),  # biases
                name='l2')

            self.v = tf.layers.dense(
                inputs=l2,
                units=1,  # output units
                activation=None,
                kernel_initializer=tf.random_normal_initializer(0.,
                                                                .1),  # weights
                bias_initializer=tf.constant_initializer(0.1),  # biases
                name='V')

        with tf.variable_scope('squared_TD_error'):
            self.td_error = self.r + GAMMA * self.v_ - self.v
            self.loss = tf.square(
                self.td_error)  # TD_error = (r+gamma*V_next) - V_eval
        with tf.variable_scope('train'):
            self.train_op = tf.train.AdamOptimizer(lr).minimize(self.loss)
```
then , after 250 episode which can get the best reward
![improve](https://user-images.githubusercontent.com/19503080/48131087-cf5bbd80-e2c9-11e8-9533-e94928de7547.jpg)

pls confirm it thanks!

**Comment by MorvanZhou** [comment]
Hi, thanks for the point.
Have you removed the random seed at the beginning of this script?

```
np.random.seed(2)
tf.set_random_seed(2)
```

**Comment by RozenAstrayChen** [comment]
No I have'nt remove the random seed

**Comment by MorvanZhou** [comment]
The performance will be affected by the random seed. Please remove it and try it again.
